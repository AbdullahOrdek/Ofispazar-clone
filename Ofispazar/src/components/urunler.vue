<template>
    <div class="arkaplan" v-if="products && products.length">
        <router-link tag="div" :to="{ name: 'urun', params: { id: product.id } }"
        v-for="product in products"
        :key="product.id" class="urunkart"> <img style="height:170px;" :src="product.image">
        <p id="detay"> {{product.urunAdi}}</p>
        

        <p id="fiyat" > {{ Number.parseFloat(product.fiyat).toFixed(2) }} TL</p></router-link>
        
    </div>
</template>
<script>
import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";
import { library } from "@fortawesome/fontawesome-svg-core";
import { fas } from "@fortawesome/free-solid-svg-icons";
library.add(fas);
import { mapActions } from "vuex";
export default {
  components: {
    FontAwesomeIcon,
  },
  data() {
    return {
      products: [],
    };
  },
  created() {
    this.fetchProducts().then((data) => {
      this.products = data;
    });
  },
  methods: {
    ...mapActions(["fetchProducts"]),
  },
};
</script>

<style scoped>
.urunkart:hover{
    opacity: 0.3;
    cursor: pointer;
}
.arkaplan{
    border: 1px solid #D3D3D3;
    display: inline-block;
    width: 42%;
    margin-left: 30px;
    margin-top: 0px;
}
.urunkart{
    height: 370px;
    width: 24.87%;
    float: left;
    border-right: 0.5px solid #D3D3D3;
    border-bottom: 0.5px solid #D3D3D3;
    background-color: white                                                             ;
}
img{
    width: 180px;
   float: left;
    margin: 10px;
    margin-bottom: 60px;
    
}
#detay{
    text-align: center;
    font-size: 12px;
    margin-bottom: 40px;
    text-decoration: none;
}
#fiyat{
   color:#c50000;
   text-align: center;
   font-size: 20px;
   margin-bottom: 40px;
   text-decoration: none !important;
   

}

</style>