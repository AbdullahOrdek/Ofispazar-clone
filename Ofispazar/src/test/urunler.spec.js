import { mount, createLocalVue } from "@vue/test-utils";
import Vuex from "vuex"
import urunler from "../components/urunler.vue";
describe("Ürünler Listesi", () => {
    const localVue = createLocalVue()
    localVue.use(Vuex)
    const mockFunction = jest.fn();
    const products = [{ "id": 1, "urunAdi": "Fotokopi Kiralama /1 Ay - Yazıcı -Tarayıcı- Faks- A3 -Network", "urunAciklama": "Utax", "fiyat": 589, "image": "https://i.hizliresim.com/hhpqxfp.jpg" },
    { "id": 2, "urunAdi": "Kyocera M3645DN Fotokopi Yazıcı Tarayıcı Faks", "urunAciklama": "Kyocera", "fiyat": 14884, "image": "https://i.hizliresim.com/fte3dku.jpg" },
    { "id": 3, "urunAdi": "Fotokopi Kiralama /1 Ay - Yazıcı -Tarayıcı- Faks- A4 -Network", "urunAciklama": "Utax", "fiyat": 399, "image": "https://i.hizliresim.com/przaxcq.jpg" },
    ];
    const fetchProducts = () => {
        return new Promise(resolve => {
            resolve(products)
        })
    };
    const wrapper = mount(urunler, {
        stubs: ["router-link"],
        localVue,
        store: new Vuex.Store({
            actions: {
                fetchProducts: fetchProducts
            },
        })
    })

    describe("Liste Kontrolü", () => {
        it("Fotokopi Kiralama /1 Ay - Yazıcı -Tarayıcı- Faks- A3 -Network :", () => {
            expect(wrapper.text()).toContain("Fotokopi Kiralama /1 Ay - Yazıcı -Tarayıcı- Faks- A3 -Network");
        });
        it("Ürün 2'nin açıklaması : Kyocera :", () => {
            expect(wrapper.text()).toContain("Kyocera");
        });

        it("Ürün 3'ün fiyatı : 399", () => {
            expect(wrapper.text()).toContain("399");
        });
        test("Snapshot Testi", () => {
            expect(wrapper.html()).toMatchSnapshot();
        });
    });
    test('fetchProducts Fonsiyonu Ürün listesini çözümler', () => {
        return expect(fetchProducts()).resolves.toBe(products);
    });
})