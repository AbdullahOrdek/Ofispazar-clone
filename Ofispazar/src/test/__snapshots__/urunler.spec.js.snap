// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ürünler Listesi Liste Kontrolü Snapshot Testi 1`] = `
"<div class=\\"arkaplan\\">
  <router-link-stub tag=\\"div\\" to=\\"[object Object]\\" class=\\"urunkart\\"><img src=\\"https://i.hizliresim.com/hhpqxfp.jpg\\" style=\\"height: 170px;\\">
    <p id=\\"detay\\"> Fotokopi Kiralama /1 Ay - Yazıcı -Tarayıcı- Faks- A3 -Network</p>
    <p id=\\"fiyat\\"> 589.00 TL</p>
  </router-link-stub>
  <router-link-stub tag=\\"div\\" to=\\"[object Object]\\" class=\\"urunkart\\"><img src=\\"https://i.hizliresim.com/fte3dku.jpg\\" style=\\"height: 170px;\\">
    <p id=\\"detay\\"> Kyocera M3645DN Fotokopi Yazıcı Tarayıcı Faks</p>
    <p id=\\"fiyat\\"> 14884.00 TL</p>
  </router-link-stub>
  <router-link-stub tag=\\"div\\" to=\\"[object Object]\\" class=\\"urunkart\\"><img src=\\"https://i.hizliresim.com/przaxcq.jpg\\" style=\\"height: 170px;\\">
    <p id=\\"detay\\"> Fotokopi Kiralama /1 Ay - Yazıcı -Tarayıcı- Faks- A4 -Network</p>
    <p id=\\"fiyat\\"> 399.00 TL</p>
  </router-link-stub>
</div>"
`;
